src/witan/app/forecast.clj

```diff
@@ -340,3 +361,31 @@
                  (s/validate
                   [ws/Forecast]
                   (map ->Forecast result)))))
+
+(defresource input-data [{:keys [id version category user-id]}]
+  util/json-resource
+  :allowed-methods #{:post}
+  :exists? (fn [ctx]
+             (let [forecast (get-forecast-version id version)
+                   category-exists (some->> forecast
+                                            :model_id
+                                            (model/get-model-by-model-id)
+                                            :input_data
+                                            (some #{category}))]
+               [category-exists {:forecast forecast}]))
+  :processable? (fn [ctx]
+                  (and (util/post!-processable-validation ws/NewDataItem)
```

Comment:
`util/post!-processable-validation` actually returns a fn so this line should be `((util/post!-processable-validation ws/NewDataItem) ctx)` - I do not recall why it was written this way....